File: examples/code/curve/FStar_Ghost.ml

open Prims

type 'Aa erased =
'Aa


let reveal = (Obj.magic ((fun x -> (Obj.magic (())))))


let hide = (fun x -> x)


let lemma_hide_reveal = (Obj.magic ((fun x -> ())))


let as_ghost = (fun f -> (f ()))


let elift1 = (Obj.magic ((fun f ga -> (f (Obj.magic ((Obj.magic (()))))))))


let elift2 = (Obj.magic ((fun f ga gc -> (f (Obj.magic ((Obj.magic (())))) (Obj.magic ((Obj.magic (()))))))))


let elift3 = (Obj.magic ((fun f ga gc gd -> (f (Obj.magic ((Obj.magic (())))) (Obj.magic ((Obj.magic (())))) (Obj.magic ((Obj.magic (()))))))))


let elift1_p = (Obj.magic ((fun f ga -> (f (Obj.magic ((Obj.magic (()))))))))


let elift2_p = (Obj.magic ((fun f ga gc -> (f (Obj.magic ((Obj.magic (())))) (Obj.magic ((Obj.magic (()))))))))


let gcut = ()


let gassume = (fun uu____234 -> ())

File: fsharp/base/FStar_Ghost.fs
module FStar_Ghost

type erased = unit
let reveal : erased -> unit = fun _ -> ()
let hide   : unit -> erased = fun  _ -> ()
let hide_reveal : erased -> unit = fun _ -> ()
let reveal_hide : unit -> unit = fun _ -> ()
let elift1   : (unit -> unit) -> erased -> erased = fun _ _ -> ()
let elift2   : (unit -> unit -> unit) -> erased -> erased -> erased = fun _ _ _ -> ()
let elift3   : (unit -> unit -> unit -> unit) -> erased -> erased -> erased -> erased = fun _ _ _ _ -> ()
let elift1_p : (unit -> unit) -> erased -> erased = fun _ _ -> ()
let elift2_p : (unit -> unit -> unit) -> erased -> erased -> erased = fun _ _ _ -> ()
